### DISCUSSION TITLE
Непонятки с коммерческим использованием в PRO лицензии

### DISCUSSION THREAD

**Discussion by simplepersonru** [totumonline/totum-ru-issues-and-discussions#374] [Неопределенное]
У меня возникло стойкое желание попробовать использовать totum для внедрения проектов за деньги. Я получил доступ к PRO и  почитал условия использования в лицензии. Вот небольшая выдержка:

1.Термины и определения:
«Использование» — означает любые действия, осуществляемые с Программным обеспечением, распространяемым на условиях настоящей Лицензии.
3.1 Пользователь вправе **использовать** или изменять Программное обеспечение только для своих внутренних деловых целей или для личного использования.

Я показал это юристу, и он очень сильно засомневался насчет коммерческого использования. Формулировка "своих внутренних деловых целей" мало похоже на коммерческое использование

В то же время я пообщался с @ttm-alexeypolunin, и он сказал, что допускается свободное коммерческое использование, но без "продажи админского доступа к PRO, установленном на вашем или клиентском сервере". Соответственно если делать облачный сервис для неограниченного круга лиц на базе totum-pro, то подразумеваются комиссионные отчисления по договоренности.

Соответственно лицензия расходится с действительностью. При большом желании, это, наверное, можно использовать как рычаги давления (рассматриваю гипотетические варианты). Очень хочется чтобы формулировки коммерческого использования, или им подобные, вошли в лицензию

Вопрос - запланируется ли дописывание лицензии под соответствие действительности?

**Comment by ttm-alexeypolunin** [comment]
«Один юрист — четыре мнения» и «Все зависит от позиции» — это вместо вступления, что бы было понятно, что юриспруденция — это не законы физики.

В конечном решении на любой платформе (тотум не исключение) есть 2 составляющие:

1. Код платформы
2. Код Конфигурации (обработки). В Тотум мы называем это схемой (schema)

На платформу в случае с Тотумом эмитентом прав являемся мы и любой ваш клиент может получить версию PRO с сопутствующим лицензионным разрешением для личного и делового использования — **для этого ему от своего аккаунта надо поставить звезду основному репо и получить доступ к PRO.**

Если со звездой сложно по идеологическим причинам — обращайтесь, выдадим лайфтайм доступ за €1К.

Лицензию на Схему и дополнительные модули написанные вами — определяете вы как разработчик.

Вот и все.

По тексту лицензии изменений не будет тк за образец взята лицензия n8n. У них на борде RunaCap с хорошим я думаю юрсоставом и основная ее идея — это cloud restricted.

**Reply by simplepersonru** [reply]
Допустим есть клиент. Я сделал **схему** под нужды клиента, развернул сервер на своих мощностях, получил деньги. Администратором системы являюсь я, у которого есть доступ к PRO. Я на своем сервере со своего github аккаунта, на котором есть доступ к totum-pro. 

Клиент чтобы **использовать** (вспоминаем лицензию) этот мой сервис с позиции пользователя, обязан иметь доступ к PRO? Притом что в сервисе будут функции, которые относятся к PRO версии. Но у клиента не будет учетки с правами создателя

**Reply by ttm-alexeypolunin** [reply]
Я бы трактовал это как ваше деловое использование. Но если вы отдали в этом случае админ-учетки вашим клиентам, что бы они дальше разрабатывали сами, а вы бы обслуживали сервер — это уже явный cloud.

Другое дело, что нам ресурсы не позволят за вами гонятся в судах (это достаточно дорого).

Все это сделано на всякий случай, если вдруг проект станет популярным, его захочет купить Гугл и выяснится, что у Майкрософт есть право использовать его так-же бесплатно.

Я утрирую, тк ни тому не другому тотум нафиг не нужен и врядли будет нужен, но на этот пункт смотрят ребята из инвестфондов.

**Reply by simplepersonru** [reply]
админ-учетка = учетная запись с системной ролью Creator ?

**Reply by ttm-alexeypolunin** [reply]
Угу

**Reply by simplepersonru** [reply]
>Но если вы отдали в этом случае админ-учетки вашим клиентам, что бы они дальше разрабатывали сами, а вы бы обслуживали сервер — это уже явный cloud

Рассмотрим вариант, когда получается явный cloud. Допустим таким cloud пользуется N клиентов. Насколько я понял из нашего диалога в telegram, здесь есть 2 варианта:
1. Каждому из этих клиентов нужно получить доступ к PRO - либо через звезду, либо за деньги, тут все понятно
2. Мы с вами, как с **лицензиаром** заключаем некие договоренности о комиссионных отчислениях и тогда cloud спокойно работает без необходимостей манипуляций со звездами и покупки PRO для каждого клиента

Условия по пункту 2 индивидуальны в каждом случае и обсуждаются отдельно или есть какая-то общая практика? Возможно какие-то существующие примеры?

**Reply by ttm-alexeypolunin** [reply]
Нет никакой практики на текущий момент. Я думаю что в этом случае лицензия должна определять, какое конкретное конечное решение вы тиражируете.  Как там считать роялти открытый вопрос. Скорее всего — это фиксированная сумма в год на лимит количество установок, так-как посчитать сколько вы заработали невозможно (это игрушки на unreal тиражируются через третьи платформы и там статистика известна) но это не точно, я не думал заранее.

**Reply by simplepersonru** [reply]
подытожим, что индивидуальны в каждом случае и обсуждаются отдельно

Спасибо за ответы!

**Reply by ttm-alexeypolunin** [reply]
Обновили лицензию для внятности того, что здесь обсуждалось: https://github.com/totumonline/totum-pro-license/blob/main/license_ru_last

Заодно закончили программу PRO за звездочки на GH.
